src/components/Calendar/Calendar.tsx

```diff
@@ -30,22 +32,40 @@ function Calendar({
   variant = 'default',
   subtitle = '',
 }: Props) {
-  const current = date || dayjs();
+  const [current, setCurrent] = useState(dates.length > 0 ? dates[0] : dayjs());
   const weeks = getWeeks(current.year(), current.month());
 
+  const select = (date: dayjs.Dayjs) => {
+    if (type === 'single') {
+      onSelect([date]);
+    } else if (type === 'range') {
+      if (dates.length === 0 || dates.length === 2) {
+        onSelect([date]);
+      } else if (dates.length === 1) {
+        onSelect(date <= dates[0] ? [date] : [...dates, date]);
```

Comment:
when comparing dates, I would propose to always use dayjs functions. It seems more expressive and maybe covers some edge cases that we maybe are missing otherwise.